### PR TITLE
Add analytical quality assurance documents

### PR DESCRIPTION
**Pull request by ESKYoung**
Add analytical quality assurance (AQA) document templates. These include:

- AQA plan
- Assumptions and caveats log
- PR template

Have also included functionality so that OGDs can include their own frameworks easily, and the PR template is placed in the correct location depending on your remote host (GitHub/GitLab).

Note the post-cookiecutter hooks require tests, and documentation is needed for the whole process, but will do these at a a later date.

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/aqa_plan.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Analytical quality assurance plan
+
+This analytical quality assurance (AQA) plan outlines our implementation of the [Aqua Book](
+https://www.gov.uk/government/publications/the-aqua-book-guidance-on-producing-quality-analysis-for-government) for
```
Comment (mammykins) [inline {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/aqa_plan.md:4]: 2015, is this the most recent version?

File: {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/assumptions_caveats.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+| GREEN | Reliable assumption, well understood and/or          | Marginal assumptions; their changes have no or        |
+|       | documented; anything up to a validated & recent set  | limited impact on the outputs.                        |
+|       | of actual data.                                      |                                                       |
++-------+------------------------------------------------------+-------------------------------------------------------+
+| AMBER | Some evidence to support the assumption; may vary    | Assumptions with a relevant, even if not critical,    |
+|       | from a source with poor methodology to a good source | impact on the outputs.                                |
+|       | that is a few years old.                             |                                                       |
++-------+------------------------------------------------------+-------------------------------------------------------+
+| RED   | Little evidence to support the assumption; may vary  | Core assumptions of the analysis; the output would be |
```
Comment (mammykins) [inline {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/assumptions_caveats.md:29]: Clear language

File: {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/assumptions_caveats.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+The log contains the following assumptions and caveats:
+
+<!-- Use reStructuredText contents directive to generate a local contents -->
+```eval_rst
+.. contents::
+    :local:
+    :depth: 1
+```
+
+### A1:
```
Comment (mammykins) [inline {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/assumptions_caveats.md:46]: Stands for Assumption 1?

File: {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Summary
+
+Add your summary here - keep it brief, to the point, and in plain English.
```
Comment (mammykins) [inline {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md:3]: Can you give a good example? or would that clutter

File: {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+-->
+
+### General checks
+
+- [ ] Code runs
+- [ ] Developments are **secure** and **ethical**
+- [ ] You have made proportionate checks that the code works correctly
+- [ ] Test suite passes
+- [ ] Assumptions, and caveats log updated (see `docs/aqa/assumptions_caveats.md`), if necessary
+- [ ] [Minimum usable documentation](http://agilemodeling.com/essays/documentLate.htm) written in the `docs` folder
```
Comment (mammykins) [inline {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md:23]: useful link

Comment (mammykins) [reply {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md:23]: not https

**Review by mammykins — APPROVED** [review]
I read the documentation and made a few comments. Perhaps a few more links might help to give the user some concrete examples of some of the language used; government links would be preferred.

The code was well structured and documented.

I didn't build a repo using this branch of govcookiecutter. I didn't test the use of a custom Github PR template (which is a nice feature btw; does a default md exist for this in this repo?)

General observation, not PR related: the cookiecutter is coming along nicely, dunno how i survived without it all these years, i particularly like the commented out examples in all the files, such as in .envrc to remind me how to use this stuff when it’s been a while.

:shipit:

File: {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/aqa_plan.md (hunk 1 above)
Comment (ESKYoung) [reply {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/aqa_plan.md:4]: Yes it is Although I've added some additional related [resources](https://www.gov.uk/government/collections/aqua-book-resources) as well in commit [`e48b054 `](https://github.com/ukgovdatascience/govcookiecutter/pull/6/commits/e48b0546bd490fb6cedc5e16aa4436ce63c91e8e)!

Marking as **Resolved**.

File: {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/assumptions_caveats.md (hunk 3 above)
Comment (ESKYoung) [reply {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/assumptions_caveats.md]: Yes, I've made that more explicit, and added some missing placeholder text in commit [`c414e4a`](https://github.com/ukgovdatascience/govcookiecutter/pull/6/commits/c414e4add0dc6d1a0103cd984bdc198e07a849ab).

Marking as **Resolved**.

File: {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md (hunk 4 above)
Comment (ESKYoung) [reply {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md:3]: Might be clutter, but also might be useful! I'm thinking something like this:

> I've written a script to make predictions from any pre-trained scikit-learn classifier using data from a Google 
> BigQuery database.
>
> This enables us to classify unseen data as it comes in, returning a binary prediction for each session. We can use 
> these predictions to segment the data for further downstream intervention.
>
> The script is written in Python, and accepts **any** scikit-learn-based classifier. It includes additional scripts to 
> load data from the database, perform simple pre-processing for the classifier, and output the predictions as a CSV 
> file.
>
> Tests have been written for all the functions in the script; these are mainly in data loading and pre-processing. I 
> have assumed that all strings are in ASCII format. Performance tests on the chosen classifier are outside of scope, 
> and it is assumed to be a correctly tuned classifier.
>
> The majority of the scripts are straightforward, but I would like the reviewer to focus mainly on the pre-processing 
> steps, as the existing ones used in training the classifier have been refactored for performance.

I can either add it as HTML comments (so you won't see it unless you're looking at the raw Markdown), or directly in Markdown.

Alternatively, to reduce clutter, I could modify the Markdown to refer to the [GDS Way](https://gds-way.cloudapps.digital/standards/pull-requests.html) on pull requests if you think that's sufficient?

File: {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md (hunk 5 above)
Comment (ESKYoung) [reply {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md:23]: Leaving this as is, because the `https` version is a bit odd?! If I can find a similar reference in future, I'll swap it out.

Marking this as **Resolved**.

File: {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+Outstanding actions should be completed before reviewers are assigned; if actions are irrelevant, please try and add a 
+comment stating why.
+
+Incomplete pull/merge requests MAY be blocked until actions are resolved, or closed at the reviewers' discretion.
+-->
+
+### General checks
+
+- [ ] Code runs
+- [ ] Developments are **secure** and **ethical**
```
[the remark this answers is not in the document]
Comment (ESKYoung) [reply {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md]: Added link in commit [`8f62424`](https://github.com/ukgovdatascience/govcookiecutter/pull/6/commits/8f624244aef617c38eb6b659256ef4875ded79ad).

Marking as **Resolved**.

File: {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/assumptions_caveats.md (hunk 2 above)
Comment (ESKYoung) [reply {{ cookiecutter.repo_name }}/docs/aqa_frameworks/GDS/assumptions_caveats.md:29]: Yes - can't take credit, so have added the source in commit [`235a0e8`](https://github.com/ukgovdatascience/govcookiecutter/pull/6/commits/235a0e85feb009e9a392a0501aaeb5c20afeba36).

Marking as **Resolved**.

File: {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md (hunk 4 above)
Comment (ESKYoung) [reply {{ cookiecutter.repo_name }}/docs/pull_merge_request_templates/GDS.md]: Bit of a cluttered approach, so have instead referenced the [GDS Way](https://gds-way.cloudapps.digital/standards/pull-requests.html) on pull requests in commit [`90fff22`](https://github.com/ukgovdatascience/govcookiecutter/pull/6/commits/90fff22ff0d429da1586d98ba2862029042ec584).

Marking as **Resolved**.

**Comment by ESKYoung** [comment]
Great thanks for the review @mammykins! Changes have been made, and all resolved. Will rebase this on the new `master` and merge in!
